### PR TITLE
9 task submissions

### PR DESCRIPTION
**Pull request by PyrinAndrii**
**Changes:**
- Add `Repo::TaskSubmission` model (task_submissions table)
- Add Index, Destroy actions for TaskSubmission 
(routes, controller actions, view)
- Add task_submission factory
- Add model, and integration test covering
- Add seeds
- Add `has_one_attached :zip_file` to `Repo::TaskSubmission`,

**NOTE:** This PR doesn't add the `ActiveStorage` configurations.

**Screenshots:**
Screenshot 2022-04-18 at 20 06 27
<hr style="border:2px solid gray"> </hr>

Screenshot 2022-04-18 at 20 06 22
<hr style="border:2px solid gray"> </hr>

Screenshot 2022-04-18 at 20 06 55

### CODE REVIEW

File: app/models/repo/task.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@ class Task < ApplicationRecord
     belongs_to :challenge
     belongs_to :dependent_task, class_name: 'Repo::Task', optional: true
 
+    has_many :task_submissions, dependent: :destroy_async
```
Comment (ochupa) [inline app/models/repo/task.rb:13]: We can add a check for this assoc to task_spec as well

Comment (PyrinAndrii) [reply app/models/repo/task.rb:13]: good catch 👍, added it here -[add associations test to task](https://github.com/DevChallengeIT/devchallenge-platform/pull/32/commits/174af6c46e51c304b165a1223cc000b55b5fe20f)

**Review by ochupa — APPROVED** [review]

**Review by Lomey — APPROVED** [review]

File: app/models/repo/task_submission.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Repo
+  class TaskSubmission < ApplicationRecord
+    belongs_to :task
+    belongs_to :member
+
+    has_one_attached :zip_file
```
Comment (sveredyuk) [inline app/models/repo/task_submission.rb:8]: can we validate `zip` extension?

File: app/views/admin/task_submissions/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      <%= t('resources.task_submissions.notes') %>
+    </th>
+    <th>
+    </th>
+  </thead>
+  <tbody>
+    <% @task_submissions.each do |task_submission| %>
+      <tr id="task_submission-<%= task_submission.id %>">
+        <td>
+          <%= link_to task_submission.member.user.email, edit_admin_challenge_member_path(task.challenge, task_submission.member) %>
```
Comment (sveredyuk) [inline app/views/admin/task_submissions/index.html.erb:22]: N+1 ?

**Review by sveredyuk — APPROVED** [review]
Good job. Just check N+1 in the logs (should be red text)

**Review by sveredyuk — APPROVED** [review]

**Pull request merged**